api/src/main/java/org/infinispan/api/async/Counter.java

```diff
@@ -0,0 +1,40 @@
+package org.infinispan.api.async;
+
+import java.util.concurrent.CompletionStage;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 13.0
+ **/
+public interface Counter {
```

Comment:
I'm affraid we need another interface to support strong counter methods:
* increment/decrementAndGet
* compareAndSet/Swap
* set

name() is missing as well.

Reply:
Yes, I'll do that.

Reply:
Done